conformance-tests/mercure.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { test, expect } from '@playwright/test';
+import { randomBytes } from "crypto";
+
+function randomString() {
+  return randomBytes(20).toString('hex');
+}
+
+test.beforeEach(async ({ page }) => await page.goto('/'));
+
+test.describe('Publish update', () => {
+  const randomStrings: string[] = [];
+  for (let i = 0; i < 6; i++)
+    randomStrings[i] = randomString();
+
+  type Data = {name: string, updateTopics: string[], topicSelectors: string[], mustBeReceived: boolean, updateID?: string, private?: true};
+
+  const dataset: Data[] = [
+    {name: 'raw string', mustBeReceived: true, updateTopics: [randomStrings[0]], topicSelectors: [randomStrings[0]]},
+    {name: 'multiple topics', mustBeReceived: true, updateTopics: [randomString(), randomStrings[1]], topicSelectors: [randomStrings[1]]},
+    {name: 'multiple topic selectors', mustBeReceived: true, updateTopics: [randomStrings[2]], topicSelectors: ['foo', randomStrings[2]]},
+    {name: 'URI', mustBeReceived: true, updateTopics: [`https://example.net/foo/${randomStrings[3]}`], topicSelectors: [`https://example.net/foo/${randomStrings[3]}`]},
+    {name: 'URI template', mustBeReceived: true, updateTopics: [`https://example.net/foo/${randomStrings[4]}`], topicSelectors: ['https://example.net/foo/{random}']},
+    {name: 'nonmatching raw string', mustBeReceived: false, updateTopics: [`will-not-match}`], topicSelectors: ['another-name']},
+    {name: 'nonmatching URI', mustBeReceived: false, updateTopics: [`https://example.net/foo/will-not-match}`], topicSelectors: ['https://example.net/foo/another-name']},
+    {name: 'nonmatching URI template', mustBeReceived: false, updateTopics: [`https://example.net/foo/will-not-match}`], topicSelectors: ['https://example.net/bar/{var}']},
+    {name: 'private raw string', mustBeReceived: false, private: true, updateTopics: [randomStrings[0]], topicSelectors: [randomStrings[0]]},
+    {name: 'private URI', mustBeReceived: false, private: true, updateTopics: [`https://example.net/foo/${randomStrings[3]}`], topicSelectors: [`https://example.net/foo/${randomStrings[3]}`]},
+    {name: 'private URI template', mustBeReceived: false, private: true, updateTopics: [`https://example.net/foo/${randomStrings[4]}`], topicSelectors: ['https://example.net/foo/{random}']},
+  ];
+
+  for (const data of dataset) {
+    test(data.name, async ({ page }) => {
+      page.on('console', msg => console.log(msg.text()));
+
+      data.updateID = `id-${JSON.stringify(data.updateTopics)}`;
+
+      const { received, contentType, status, body } = await page.evaluate(async (data) => {
+        const receivedResult = Symbol('received');
+        const notReceivedResult = Symbol('not received');  
+
+        let resolveReady: Function;
+        const ready = new Promise((resolve) => {
+          resolveReady = function () {
+            resolve(true);
+          };
+        });
```

Comment:
```suggestion
        let resolveReady: () => void;
        const ready = new Promise((resolve) => {
          resolveReady = () => resolve(true)
        });
```
Minor type and syntax update, can be applied to `resolveReceived` below too if you like it